src/ConfigPostProcessor.php

```diff
@@ -265,19 +260,79 @@ private function replaceDependencyConfiguration(array $config)
      *
      * In this case, we want to keep the alias as-is, but rewrite the target.
      *
-     * This same logic can be used for invokables, which are essentially just
-     * an alias map.
+     * We need also provide an additional alias if the alias key is a legacy class.
      *
      * @return array
      */
     private function replaceDependencyAliases(array $aliases)
     {
         foreach ($aliases as $alias => $target) {
-            $aliases[$alias] = $this->replacements->replace($target);
+            $newTarget = $this->replacements->replace($target);
+            $newAlias  = $this->replacements->replace($alias);
+
+            if ($newAlias === $alias) {
+                $aliases[$alias] = $newTarget;
+                continue;
+            }
+
+            if (isset($aliases[$newAlias])) {
+                continue;
+            }
+
+            $notIn = [$newTarget];
+            $name  = $newTarget;
+            while (isset($aliases[$name])) {
+                $notIn[] = $aliases[$name];
+                $name    = $aliases[$name];
+            }
+
+            if (! in_array($newAlias, $notIn, true)) {
+                $aliases[$alias]    = $newAlias;
+                $aliases[$newAlias] = $newTarget;
+            }
         }
+
         return $aliases;
     }
 
+    /**
+     * Rewrite dependency invokables array
+     *
+     * In this case, we want to keep the alias as-is, but rewrite the target.
+     *
+     * We need also provide an additional alias if invokable is defined with
+     * an alias which is a legacy class.
+     *
+     * @return array
+     */
+    private function replaceDependencyInvokables(array $config)
+    {
+        if (empty($config['invokables'])) {
+            return $config;
+        }
+
+        foreach ($config['invokables'] as $alias => $target) {
+            $newTarget = $this->replacements->replace($target);
+
+            if ($alias !== $target) {
+                $newAlias = $this->replacements->replace($alias);
+                $config['invokables'][$newAlias] = $newTarget;
+
+                if ($newAlias !== $alias) {
+                    if (! isset($config['aliases'][$newAlias])) {
+                        $config['aliases'][$alias] = $newAlias;
+                    }
+
+                    unset($config['invokables'][$alias]);
+                }
+            } else {
+                $config['invokables'][$alias] = $newTarget;
+            }
```

Comment:
```suggestion
            if ($alias === $target) {
                $config['invokables'][$alias] = $newTarget;
                continue;
            }

            $newAlias = $this->replacements->replace($alias);
            $config['invokables'][$newAlias] = $newTarget;

            if ($newAlias === $alias) {
                continue;
            }

            if (! isset($config['aliases'][$newAlias])) {
                $config['aliases'][$alias] = $newAlias;
            }

            unset($config['invokables'][$alias]);
```

Basically, continue early, to reduce nesting.